src/pymovements/dataset/dataset_files.py

```diff
@@ -61,33 +61,45 @@ def scan_dataset(definition: DatasetDefinition, paths: DatasetPaths) -> pl.DataF
         If an error occurred during matching filenames or no files have been found.
     """
     # Get all filepaths that match regular expression.
-    fileinfo_dicts = []
-    if definition.has_gaze_files:
+    _fileinfo_dicts = {}
+    if definition.has_files['gaze']:
         fileinfo_dicts = match_filepaths(
             path=paths.raw,
-            regex=curly_to_regex(definition.filename_format),
+            regex=curly_to_regex(definition.filename_format['gaze']),
             relative=True,
         )
-    if definition.has_precomputed_event_files:
+        if not fileinfo_dicts:
+            raise RuntimeError(f'no matching files found in {paths.raw}')
+
+        fileinfo_df = pl.from_dicts(data=fileinfo_dicts, infer_schema_length=1)
```

Comment:
are we sure, that a length of 1 is enough to infer the schema?

Reply:
this is not actually refactored and I'd leave it currently as is